walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,82 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.LinkedList;
+import java.util.Comparator;
 
 public final class FindMeetingQuery {
+  List<TimeRange> busyTimes = new ArrayList<>();
+  List<TimeRange> freeTimes = new ArrayList<>();
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    busyTimes = getBusyTimes(events, request);
+    List<TimeRange> overlappingTimes = findOverlappingTimes();
+    return (getFreeTimes(overlappingTimes, request));
```

Comment:
nit: You shouldn't need the extra variable here (overlappingTimes). You can simple call findOverlappingTimes() as a parameter to getFreeTimes().